File: 0020-valid-parentheses/20 Valid Parentheses/20valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        bool valid = false;
        stack<char> brackets;
        brackets.push('a');
        brackets.push(s[0]);
        for (int i = 1; i < s.length(); i++) {
            if (brackets.size() != 0) {
                char x = brackets.top();
                brackets.push(s[i]);
                if (x == '(' && brackets.top() == ')') {
                    brackets.pop();
                    brackets.pop();
                } else if (x == '{' && brackets.top() == '}') {
                    brackets.pop();
                    brackets.pop();
                } else if (x == '[' && brackets.top() == ']') {
                    brackets.pop();
                    brackets.pop();
                }
            }
        }
        if (brackets.size() == 1) {
            valid = true;
        }
        return valid;
    }
};

File: 20 Valid Parentheses/20valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<Character>();
        st.push('0');
        st.push(s.charAt(0));

        for (int i = 1; i < s.length(); i++) {
            char x = st.peek();
            st.push(s.charAt(i));
            
            if (x == '(' && st.peek() == ')') {
                st.pop();
                st.pop();
            } else if (x == '{' && st.peek() == '}') {
                st.pop();
                st.pop();
            } else if (x == '[' && st.peek() == ']') {
                st.pop();
                st.pop();
            }
        }

        if (st.size() == 1) {
            return true;
        }

        return false;
    }
}